### PR TITLE
Pyarn add - take 2 - initial refactoring

### PR DESCRIPTION
**Pull request by lukebatchelor**
This is the initial refactoring for pyarn add.

It add yarn.add and symlinkPackageDependencies.

TODO: 
* symlinkPackage**s**Dependencies
* refactor install
* tests for install
* pyarn [w|ws|p] add + tests

### CODE REVIEW

File: src/utils/__tests__/addDependenciesToPackages.test.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
+// @flow
+
+import { getFixturePath } from 'jest-fixtures';
```
Comment (lukebatchelor) [inline src/utils/__tests__/addDependenciesToPackages.test.js]: This util is not finished yet. I realised I needed to have all of symlinking working first.

File: src/utils/addDependenciesToPackages.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+// @flow
+
```
Comment (lukebatchelor) [inline src/utils/addDependenciesToPackages.js]: Unfinished, as it was waiting on symlinking

File: src/utils/yarn.js [hunk 3]
```diff
@@ -1,10 +1,45 @@
...
+      spawnArgs.push(dep.name);
+    }
+  });
+
+  if (type) {
+    const typeToFlagMap = {
+      dependencies: '',
+      devDependencies: '--dev',
+      peerDependencies: '--peer',
+      optionalDependencies: '--optional'
```
Comment (thejameskyle) [inline src/utils/yarn.js]: I think it'd be better to have this conversion happened up in the commands rather than in the utilities

Comment (lukebatchelor) [reply src/utils/yarn.js]: I thought we said in the last PR that the utility should convert js options to cli options and commands should be converting from flags to js options?
https://github.com/pyarn/pyarn/pull/17/files#r138537987

Comment (thejameskyle) [reply src/utils/yarn.js]: Ohhh, I read this code backwards, I thought you were converting flags to types not types to flags

File: src/utils/addDependenciesToPackages.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import * as logger from './logger';
+import * as yarn from './yarn';
+
+export default async function addDependenciesToPackage(
+  project: Project,
+  pkg: Package,
+  dependencies: Array<Dependency>,
+  type?: configDependencyType
+) {
+  const isProjectPackage = project.pkg.dir === pkg.dir;
```
Comment (thejameskyle) [inline src/utils/addDependenciesToPackages.js]:
```js
pkg.isSamePackage(project.pkg)
```

File: src/utils/yarn.js (hunk 3 above)
Comment (lukebatchelor) [reply src/utils/yarn.js]: Yea, I might refactor using Keys later https://flow.org/en/docs/types/utilities/#toc-keys

File: src/utils/addDependenciesToPackages.js (hunk 4 above)
Comment (lukebatchelor) [reply src/utils/addDependenciesToPackages.js]: Ah yea,  I was waiting for the remove PR.

File: src/commands/__tests__/remove.test.js [hunk 5]
```diff
@@ -55,8 +55,8 @@ describe('pyarn remove', () => {
     ).rejects.toBeInstanceOf(Error);
   });
 
-  test('removing a dependency that is used by a workspace', async () => {
-    let { tempDir } = await copyFixtureIntoTempDir(
+  test.only('removing a dependency that is used by a workspace', async () => {
```
Comment (thejameskyle) [inline src/commands/__tests__/remove.test.js]: `.only`

**Pull request closed without merging**